manifests/params.pp

```diff
@@ -58,4 +60,13 @@
       fail("Module ${module_name} is not supported on osfamily '${::osfamily}'")
     }
   }
+	case $::architecture {
+    'x86_64', 'amd64': { $arch = 'amd64' }
+    'i386':            { $arch = '386'   }
+    /^arm.*/:          { $arch = 'arm'   }
+    default:           {
+      fail("Unsupported kernel architecture: ${::architecture}")
```

Comment:
That's fine, it's definitely an unlikely scenario so not worth introducing the complexity for it.